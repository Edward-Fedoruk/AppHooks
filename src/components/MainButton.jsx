import React from "react"
import PropTypes from "prop-types"
import Button from "@material-ui/core/Button"
import { withStyles } from "@material-ui/core"
import classNames from "classnames"

const styles = ({ breakpoints }) => ({
  btn: {
    textTransform: "capitalize",
    color: "#fff",
    fontSize: "16px",
    background: "#35C1CE",

    "&:hover": {
      opacity: ".9",
      background: "#192B81",
    },

    [breakpoints.down(425)]: {
      fontSize: "12px",
      padding: "8px 7px"
    },
  },
})

const MainButton = props => {
  const { classes, className, children, ...rest } = props
  return (
    <Button 
      size="large"
      color="primary" 
      variant="text"
      className={classNames(classes.btn, className)}
      {...rest}
    >
      {children}
    </Button>
  )
}

MainButton.propTypes = {
  classes: PropTypes.object.isRequired, 
  className: PropTypes.string, 
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
}

export default withStyles(styles)(MainButton)
