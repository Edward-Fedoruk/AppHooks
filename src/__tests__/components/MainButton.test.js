import "jsdom-global/register"
import React from "react"
import { shallow, mount } from "enzyme"
import MainButton from "../../components/MainButton"

describe("MainButton", () => {
  it("should match snapshot", () => {
    const tree = shallow(
      <MainButton />
    )
    expect(tree).toMatchSnapshot()
  })
  it("render text", () => {
    const wrapper = mount(
      <MainButton>test</MainButton>
    )
    expect(wrapper.text()).toBe("test")
  })
})
