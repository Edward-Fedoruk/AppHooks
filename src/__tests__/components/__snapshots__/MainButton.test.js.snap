// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainButton should match snapshot 1`] = `
<MainButton
  className=""
  classes={
    Object {
      "btn": "MainButton-btn-1",
    }
  }
/>
`;
